PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/layers/datalink.cpp

```diff
@@ -1,5 +1,91 @@
 #include "datalink.h"
 
+/**
+ * @brief Updates the shared credit limit values based on the received DLLP information
+ * @param flit Flit containing the DLLPs
+ * @param P_SHARED_CREDIT_LIMIT Array of shared credit limit for posted traffic
+ * @param NP_SHARED_CREDIT_LIMIT Array of shared credit limit for non-posted traffic
+ * @param CPL_SHARED_CREDIT_LIMIT Array of shared credit limit for completion traffic
+ * @param FI1 First indication (FI1) flag for credit update
+ * @param FI2 Second indication (FI2) flag for credit update
+*/
+void DatalinkLayer::updateCreditLimit(Flit flit, int(&P_SHARED_CREDIT_LIMIT)[2], int(&NP_SHARED_CREDIT_LIMIT)[2], int(&CPL_SHARED_CREDIT_LIMIT)[2], bool& FI1, bool& FI2) {
```

Comment:
If I remember correctly, array names are pointers, and therefore will be modified normally inside the function and don't need to be passed by reference manually like other data types. Please check and modify if so